Assets/ModScripts/Submodules/WhosWho.cs

```diff
@@ -0,0 +1,293 @@
+    int listB_Index = 0;
+
+    public WhosWho(CruelModkitScript Module, int ModuleID, ComponentInfo Info, byte Components) : base(Module, ModuleID, Info, Components)
+    {
+        UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] Solving Who's Who.", ModuleID);
+        LEDtoUse = DetermineLED();
+        LEDcolor = Module.LED[LEDtoUse].transform.Find("LEDL").GetComponentInChildren<Renderer>().material.name.ToLowerInvariant();
+        LEDcolor = LEDcolor.Substring(0, LEDcolor.IndexOf(' ')); // because the name has " (instance)" at the end
+        UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] The chosen LED is {1}; the offset is {2} and the row to begin with is {3}.", ModuleID, LEDcolor, offsets[LEDcolor] > 0 ? "+" + offsets[LEDcolor] : "" + offsets[LEDcolor], rows[LEDcolor] + 1);
+        int row = rows[LEDcolor];
+        int column = LEDtoUse;
+        string cellWord, cellColor;
+        for (int i = 0; i < 10; i++)
+        {
+            cellWord = words[row][column];
+            cellColor = colors[row][column];
+            if (listA.Contains(cellWord))
+                break;
+            else
+                listA.Add(cellWord);
+            row += 8;
+            row += offsets[LEDcolor];
+            row %= 8;
+            column += 8;
+            column += offsets[cellColor];
+            column %= 8;
+        }
+        string word;
+        int listBLength = Random.Range(10, 16);
+        do
+        {
+            word = ComponentInfo.WordList.PickRandom();
+            if (!listB.Contains(word))
+                listB.Add(word);
+        } while (listB.Count < listBLength);
+        UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] List A is as follows: [{1}].", ModuleID, string.Join(", ", listA.ToArray()));
+        UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] List B is as follows: [{1}].", ModuleID, string.Join(", ", listB.ToArray()));
+        Module.WidgetText[1].text = listB.First();
+        string commonWord = listB.FirstOrDefault(w => listA.Contains(w));
+        if (commonWord == null)
+        {
+            UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] Neither list has anything in common; the number to submit is 0.", ModuleID);
+            finalNumber = 0;
+        } else
+        {
+            UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] The first word that each list has in common is {1}.", ModuleID, commonWord);
+            int index = ComponentInfo.WordList.IndexOf(x => x == commonWord);
+            index = (((index / 8) + 1) + ((index % 8) + 1)) % 10;
+            finalNumber = index;
+            UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] The number to submit is {1}.", ModuleID, finalNumber);
+        }
+    }
+
+    private int DetermineLED()
+    {
+        string[] warmColors = new string[]{"red", "orange", "yellow"};
+        string[] LEDcolors = Module.LED.Select(l => l.transform.Find("LEDL").GetComponentInChildren<Renderer>().material.name).ToArray();
```

Comment:
Use Info.LED instead, and use a dictionnary to translate ints to what you want. Depeding on a material's name seems pretty shaky, what if it gets renamed?